### PR TITLE
Add meta test for new diff logic

### PR DESCRIPTION
**Pull request by hauntsaninja**
Follow up to #16112

### CODE REVIEW

**Comment by hauntsaninja** [comment]
cc @ikonst :-)

File: mypy/test/meta/test_diff_helper.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        render_diff_range(actual_ranges, actual, output=output)
+        assert output.getvalue() == "  goodbye (diff)\n  world\n"
+
+        expected = ["a", "b", "c", "d", "e", "f", "g", "h", "circle", "i", "j"]
+        actual = ["a", "b", "c", "d", "e", "f", "g", "h", "square", "i", "j"]
+
+        expected_ranges, actual_ranges = diff_ranges(expected, actual)
+
+        output = io.StringIO()
+        render_diff_range(expected_ranges, expected, output=output, indent=0)
```
Comment (ikonst) [inline mypy/test/meta/test_diff_helper.py:26]: I'veb lately acquired a taste for the arrange-act-assert style. Might be a good fit here given how similar the two methods are named, and how you're using testing terminology ("expected") in a meta fashion.

File: mypy/test/meta/test_diff_helper.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    def test_diff_ranges(self) -> None:
+        a = ["hello", "world"]
+        b = ["hello", "world"]
+
+        assert diff_ranges(a, b) == (
+            [(0, 0), (0, 2), (2, 2), (2, 2)],
+            [(0, 0), (0, 2), (2, 2), (2, 2)],
+        )
+
+        a = ["hello", "world"]
```
Comment (ikonst) [inline mypy/test/meta/test_diff_helper.py:41]: nit - could be two tests just as well

File: mypy/test/helpers.py [hunk 3]
```diff
@@ -70,7 +70,12 @@ def diff_ranges(
 
 
 def render_diff_range(
-    ranges: list[tuple[int, int]], content: list[str], colour: str | None = None
+    ranges: list[tuple[int, int]],
+    content: list[str],
+    *,
+    colour: str | None = None,
+    output: IO[str] = sys.stderr,
```
Comment (ikonst) [inline mypy/test/helpers.py:77]: Could leverage capsys :) but easier testability is also nice and less magic

**Review by ikonst — APPROVED** [review]

File: mypy/test/meta/test_diff_helper.py (hunk 1 above)
Comment (hauntsaninja) [reply mypy/test/meta/test_diff_helper.py:26]: Wasn't aware of that term. I think I have a little blurring between act and assert, but maybe that's okay? I am asserting two different things with similar "arrangement"

Comment (ikonst) [reply mypy/test/meta/test_diff_helper.py:26]: As a general rule, "# Act" is where you call the function under test. I think it's ok to assert multiple things, probably just try not to eg. define expectations ahead of the "# Assert"  section.

**Pull request merged**